gcsfs/core.py

```diff
@@ -330,16 +330,24 @@ def base(self):
     def project(self):
         return self.credentials.project
 
+    # Clean up the aiohttp session
+    #
+    # This can run from the main thread if invoked via the weakref callbcak.
+    # This can happen even if the `loop` parameter belongs to another thread
+    # (e.g. the fsspec IO worker). The control flow here is intended to attempt
+    # in-thread asynchronous cleanup first, then fallback to synchronous
+    # cleanup (which can handle cross-thread calls).
     @staticmethod
     def close_session(loop, session):
         if loop is not None and session is not None:
             if loop.is_running():
                 try:
-                    loop = asyncio.get_event_loop()
-                    loop.create_task(session.close())
+                    current_loop = asyncio.get_event_loop()
+                    current_loop.create_task(session.close())
```

Comment:
Yeah. From what I can tell

1. The GC weakref callback runs in the main thread.
1. The main thread has no event loop, so when we call `get_event_loop()` it fails because it creates a new one which has never been started.
1. `create_task()` then fails because it refuses to create a task on a closed loop, and `BaseEventLoop._check_closed()` raises the exception.

With this fix, we don't see the exception anymore, but we do see the following, which is related but I don't think is caused by this change.

```
RuntimeWarning: coroutine 'ClientSession.close' was never awaited
```